packages/@aws-cdk/core/lib/bundling.ts

```diff
@@ -53,6 +53,12 @@ export interface BundlingOptions {
    * @default - uid:gid of the current user or 1000:1000 on Windows
    */
   readonly user?: string;
+
+  /**
+   * A user function to indicate if bundling should run locally instead
+   * of running in a Docker container.
+   */
+  runLocally?(): boolean;
```

Comment:
I don't think this API is supported by jsii and I would not want `BundlingOptions` to be an interface instead of a struct.

How about something like this:

```ts
export interface BundlingOptions {
  readonly docker: DockerBundling; // required
  readonly local?: ILocalBundling;
}

export interface DockerBundling {
  readonly image: BundlingDockerImage;
  readonly command?: string[];
  readonly volumes?: DockerVolume[];
  readonly environment?: { [key: string]: string; };
  readonly workingDirectory?: string;
  readonly user?: string;
}

interface ILocalBundling {
  /**
   * This method is called before attempting docker bundling to allow the bundler to be executed locally.
   * If the local bundler exists, and bundling was performed locally, return `true`. Otherwise, return `false`.
   */
  tryBundle(): boolean;
}
```

Then, anyone can provide an object that implements `ILocalBundling` and implements `tryBundle()`.

Reply:
I feel like there might be some undesired implicit assumptions here:
1. Local execution uses the exact same command as the one executed in the docker image
2. The program exists in `PATH`.

In the parcel case, you are checking if the parcel installed through `node_modules` has the right version but does that necessarily mean that this is the one that will be picked up from `PATH`? 

The `ILocalBundling` and `tryBundle()` approach basically allows complete control over how the local execution is implemented. That's why I also grouped the existing options under `DockerBundling`.

The API reads: "to configure bundling, you have to specify the docker bundling commands and optionally, you can supply a method that will attempt to do local bundling".

Reply:
got it